packages/contracts-bedrock/scripts/hash-onion/GenerateHashOnion.t.sol

```diff
@@ -0,0 +1,42 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.25;
+
+import { GenerateHashOnion } from "./GenerateHashOnion.s.sol";
+import { Test } from "forge-std/Test.sol";
+
+contract GenerateHashOnionForTest is GenerateHashOnion {
+    /// @notice Override the tokens path to the test mock tokens json file.
+    /// @notice The test mock tokens json file is used for testing purposes only.
+    function _tokensPath() internal view override returns (string memory _path) {
+        _path = string.concat(vm.projectRoot(), "/scripts/hash-onion/test-mock-tokens.json");
+    }
+}
+
+contract GenerateHashOnion_Test is Test {
+    bytes32 internal constant INITIAL_ONION_LAYER = 0x290decd9548b62a8d60345a988386fc84ba6bc95484008f6362f93160ef3e563;
+
+    GenerateHashOnionForTest internal script;
+
+    function setUp() public {
+        script = new GenerateHashOnionForTest();
+    }
+
+    /// @notice Test the script reads and parses properly the tokens json file, and hash onion generation.
+    function test_generateHashOnion() public {
+        // Calculate the hash onion using the same values than given tokens file.
+        bytes32 _innerLayer = INITIAL_ONION_LAYER;
+        // Use `_i + 2` so the addresses goes incremental per pair and they are never repeated.
+        for (uint256 _i; _i < 10; _i += 2) {
+            _innerLayer = keccak256(
+                abi.encodePacked(_innerLayer, abi.encodePacked(address(uint160(_i)), address(uint160(_i + 1))))
```

Comment:
This only works with the json you provided. It's not a generic test. I would leave a comment here saying that exactly or try to make it general.
Not sure if there is a way of doing fuzzing and generating a json out of it (then running the script on that json), or having a script that creates a random json.

Reply:
I wanted to have it so the `test-mock-tokens.json` file remains in the repo and can be seen as reference for when someone wants to use it.
But I agree with you, is a good idea to have another fuzzed test in which I write to the json the values of the array.
Will add it and add more comments.